city_scrapers/spiders/il_sex_offender_management.py

```diff
@@ -0,0 +1,249 @@
+            if "WebEx" in link_text:
+                yield scrapy.Request(
+                    link.attrib["href"], callback=self._parse_meta_video
+                )
+
+    def _parse_meta_video(self, response):
+        """Parse through meta of response in JSON if link provided rather than PDF"""
+        meta_text = response.css(
+            "[type='application/json']:not(#bootstrapData)::text"
+        ).extract_first()
+        if not meta_text:
+            return
+        meta_text = meta_text.strip()
+        clean_text_vid = json.loads(meta_text)
+
+        start_time, end_time = self._video_start_end_time(clean_text_vid)
+
+        meeting = Meeting(
+            title="SEX OFFENDER MANAGEMENT BOARD",
+            description="",
+            classification=BOARD,
+            start=start_time,
+            end=end_time,
+            all_day=False,
+            time_notes="",
+            location={
+                "address": response.urljoin(""),
+                "name": "WebEx Meeting (See meeting links to confirm)",
+            },
+            links={"title": "WebEx Meeting Link", "href": response.urljoin("")},
+            source=self.start_urls[0],
+        )
+        meeting["status"] = self._get_status(meeting)
+        meeting["id"] = self._get_id(meeting)
+
+        yield meeting
+
+    def _video_start_end_time(self, clean_text_vid):
+        """Parse through JSON and return start, end naive datetime objects"""
+
+        # Get date time formatted as '2020-09-16 13:00'
+        date_time = clean_text_vid["meetingData"]["startTime"][:-3]
+
+        # get meeting duration in a list of [hours,minutes]
+        scheduled_duration = clean_text_vid["meetingData"]["scheduledDuration"]
+
+        # create start and end time datetime naive objects
+        start_time = datetime.strptime("{}".format(date_time), "%Y-%m-%d %H:%M")
+
+        end_time = start_time + timedelta(minutes=scheduled_duration)
+
+        return start_time, end_time
+
+    def _parse_documents(self, response):
+        """Parse meeting information from agenda PDF"""
+
+        links_dict = {
+            "title": "Meeting Agenda",
+            "href": response.urljoin(""),
+        }
+        clean_text = self._cleanUpPDF(response)
+
+        if self._parse_status(clean_text) == CANCELLED:
+            meeting = Meeting(
+                title="SEX OFFENDER MANAGEMENT BOARD",
+                description="",
+                classification=BOARD,
+                start=self._parse_start(clean_text),
+                end=self._parse_end(clean_text),
+                all_day=False,
+                time_notes="See agenda to confirm details",
+                location={"address": "", "name": "Meeting Cancelled"},
+                links=links_dict,
+                source=self.start_urls[0],
+            )
+            meeting["status"] = self._get_status(meeting, text="Meeting is cancelled")
```

Comment:
Instead of hard-coding text to pass, the `_get_status` function is already checking whether any form of cancellation shows up in the text, so we can pass `clean_text` in the `text` kwarg.